File: app/src/services/adminService.js
import api from '../config/api';

export const adminService = {
  getDashboardStats: async () => {
    const response = await api.get('/admin/dashboard');
    return response.data;
  },

  getAllUsers: async (params = {}) => {
    const response = await api.get('/admin/users', {params});
    return response.data;
  },

  updateUserStatus: async (userId, isActive) => {
    const response = await api.put(`/admin/users/${userId}/status`, {
      isActive,
    });
    return response.data;
  },

  getAllSalons: async (params = {}) => {
    const response = await api.get('/admin/salons', {params});
    return response.data;
  },

  approveSalon: async (salonId, isApproved) => {
    const response = await api.put(`/admin/salons/${salonId}/approve`, {
      isApproved,
    });
    return response.data;
  },

  updateSalonStatus: async (salonId, isActive) => {
    const response = await api.put(`/admin/salons/${salonId}/status`, {
      isActive,
    });
    return response.data;
  },

  getAllBookings: async (params = {}) => {
    const response = await api.get('/admin/bookings', {params});
    return response.data;
  },

  getRevenueAnalytics: async (startDate, endDate) => {
    const response = await api.get('/admin/revenue', {
      params: {startDate, endDate},
    });
    return response.data;
  },
};

File: luxtril_flutter/lib/services/admin_service.dart
import 'api_client.dart';
import '../models/salon.dart';
import '../models/user.dart';

class AdminService {
  final ApiClient _api = ApiClient();

  Future<List<Salon>> getAllSalons({String? status}) async {
    final params = <String, String>{};
    if (status != null && status != 'all') params['status'] = status;
    final response = await _api.get('/admin/salons', queryParams: params.isNotEmpty ? params : null);
    final List<dynamic> data = response['salons'] ?? response['data'] ?? [];
    return data.map((s) => Salon.fromJson(s)).toList();
  }

  Future<Salon> approveSalon(String id) async {
    final response = await _api.put('/admin/salons/$id/approve');
    return Salon.fromJson(response['salon'] ?? response);
  }

  Future<Salon> rejectSalon(String id, {String? reason}) async {
    final response = await _api.put('/admin/salons/$id/reject', body: {
      if (reason != null) 'reason': reason,
    });
    return Salon.fromJson(response['salon'] ?? response);
  }

  Future<List<User>> getAllUsers() async {
    final response = await _api.get('/admin/users');
    final List<dynamic> data = response['users'] ?? response['data'] ?? [];
    return data.map((u) => User.fromJson(u)).toList();
  }

  Future<void> deleteUser(String id) async {
    await _api.delete('/admin/users/$id');
  }
}
